src/main/AllocatorSuperblock.cpp

```diff
@@ -0,0 +1,307 @@
+/*
+ *  R : A Computer Language for Statistical Data Analysis
+ *  Copyright (C) 1995, 1996  Robert Gentleman and Ross Ihaka
+ *  Copyright (C) 1998-2007   The R Development Core Team.
+ *  Copyright (C) 2008-2014  Andrew R. Runnalls.
+ *  Copyright (C) 2014 and onwards the Rho Project Authors.
+ *
+ *  Rho is not part of the R project, and bugs and other issues should
+ *  not be reported via r-bugs or other R project channels; instead refer
+ *  to the Rho website.
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, a copy is available at
+ *  https://www.R-project.org/Licenses/
+ */
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <cstdint>
+#include <cstdlib>
+#include <functional>
+#include <limits>
+#include <memory>
+
+#include "rho/AddressSanitizer.hpp"
+#include "rho/AllocationTable.hpp"
+#include "rho/AllocatorSuperblock.hpp"
+
+// Arena for small-object superblocks:
+uintptr_t arena_superblock_start = 0;
+uintptr_t arena_superblock_end = 0;
+uintptr_t arena_superblock_next = 0;
+
+void rho::AllocatorSuperblock::allocateArena() {
+  void* arena = nullptr;
+  size_t space = s_arenasize; // Total acquired arena space.
+  posix_memalign(&arena, s_small_superblock_size, space);
+#ifdef HAVE_STD_ALING
+  if (!arena) {
+    // Try to use operator new and std::align()  instead.
+    arena = new char[s_arenasize];
+    std::align(s_small_superblock_size,
+        s_small_superblock_size,
+        &arena,
+        &space);
+  }
+#endif // HAVE_STD_ALING
+  if (!arena) {
+    allocerr("failed to allocate small-object arena");
+  }
+  size_t num_superblock = space / s_small_superblock_size;
+  arena_superblock_start = reinterpret_cast<uintptr_t>(arena);
+  arena_superblock_end = arena_superblock_start
+      + num_superblock * s_small_superblock_size;
+  arena_superblock_next = arena_superblock_start;
+
+#ifdef HAVE_ADDRESS_SANITIZER
```

Comment:
There's no need for the #ifdef statements here.  ASAN_POISON_MEMORY_REGION is a no-op if asan isn't enabled.